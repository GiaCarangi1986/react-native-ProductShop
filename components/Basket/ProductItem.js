import React from "react";
import { Text, CheckBox, View, Button } from 'react-native';
import Wrapper from "../../views/Wrappers";
import style from "./style";

const ProductItem = ({ title = '', cost = 0, unit = 'р', icon = 'Изменить кол-во', formik = {},
  topRadius = null, bottomRadius = null, id = -1 }) => {
  const handleChange = (e) => {
    const obj = formik.values[id]
    obj.id = e
    formik.setFieldValue([id], obj)
  }

  return (
    <Wrapper nameOfStyle='card-product' topRadius={topRadius} bottomRadius={bottomRadius}>
      <CheckBox
        onValueChange={handleChange}
        value={formik.values[id]?.id}
        style={style.checkBox}
      />
      <View>
        <Text>{title}</Text>
        <View style={style.price}>
          <Text>{cost}</Text>
          <Text>{unit}</Text>
        </View>
      </View>
      <View style={style.action}>
        <Text style={style.product_detail}>{formik.values[id]?.count}</Text>
        <Text>{icon}</Text>
        {/* <View>
          <Button></Button>
        </View> */}
      </View>
    </Wrapper>
  )
}

export default ProductItem