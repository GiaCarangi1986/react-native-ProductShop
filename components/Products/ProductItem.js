import React from "react";
import { Text } from 'react-native';
import { Button } from 'react-native-elements';
import Wrapper from "../../views/Wrappers";
import { add_to_basket_icon } from '../../assets'
import Icon from '../../views/Icon'
import style from "./style";

const ProductItem = ({ title = '', cost = 0, unit = 'Ñ€', formik = {}, id = -1, no_margin = null, setModalVisible = () => { } }) => {
  const handleChangeCountPlus = () => {
    const obj = formik.values[id]
    obj.count++
    formik.setFieldValue([id], obj)
  }

  const handleChangeCountMinus = () => {
    const obj = formik.values[id]
    if (obj.count > 1) {
      obj.count--
    }
    formik.setFieldValue([id], obj)
  }
  return (
    <Wrapper nameOfStyle='card-product' otherStyle='card-product_list-products' no_margin={no_margin}>
      <Wrapper nameOfStyle='horizontal-container_product' otherStyle='horizontal-container_list-products'>
        <Button
          icon={
            <Icon
              src={add_to_basket_icon}
            />
          }
          buttonStyle={style.no_padding}
          type="clear"
          onPress={() => setModalVisible(true)}
        />
        <Wrapper nameOfStyle='check-product'>
          <Text style={style['product_title']}>{title}</Text>
          <Wrapper nameOfStyle='horizontal-container'>
            <Text>{cost}</Text>
            <Text>{unit}</Text>
          </Wrapper>
        </Wrapper>
      </Wrapper>
      <Wrapper >
        <Text style={style.product_detail}>{formik.values[id]?.count}</Text>
        <Wrapper nameOfStyle='change-count'>
          <Button title='-' buttonStyle={style.btn} titleStyle={style.btn_text} type="outline" onPress={handleChangeCountMinus} disabled={formik.values[id]?.count === 1} />
          <Button title='+' buttonStyle={style.btn} titleStyle={style.btn_text} type="outline" onPress={handleChangeCountPlus} />
        </Wrapper>
      </Wrapper>
    </Wrapper>
  )
}

export default ProductItem