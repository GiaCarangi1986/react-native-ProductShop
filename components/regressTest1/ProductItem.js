import React from "react";
import { Text, CheckBox, Button, View } from 'react-native';
import style from "./style";

export const ProductItem = ({ title = '', cost = 0, unit = 'Ñ€', formik = {}, topRadius = null, bottomRadius = null, id = -1,
  updateProduct = () => { }, max = 1 }) => {

  const handleChangeCost = (e) => {
    const obj = formik.values[id]
    obj.id = e
    formik.setFieldValue([id], obj)
  }

  const handleChangeCountPlus = () => {
    const obj = formik.values[id]
    if (obj.count <= max) {
      obj.count++
    }
    formik.setFieldValue([id], obj)
    updateProduct(id, 1)
  }

  const handleChangeCountMinus = () => {
    const obj = formik.values[id]
    if (obj.count > 1) {
      obj.count--
    }
    formik.setFieldValue([id], obj)
    updateProduct(id, -1)
  }

  return (
    <View nameOfStyle='card-product' topRadius={topRadius} bottomRadius={bottomRadius}>
      <View >
        <CheckBox
          id='checkbox'
          onValueChange={handleChangeCost}
          value={formik.values[id]?.id}
        />
        <View nameOfStyle='check-product'>
          <Text style={style['product_title']}>{title}</Text>
          <View >
            <Text>{cost}</Text>
            <Text>{unit}</Text>
          </View>
        </View>
      </View>
      <View >
        <Text style={style.product_detail}>{formik.values[id]?.count}</Text>
        <View nameOfStyle='change-count'>
          <Button id='minus' title='-' buttonStyle={style.btn} titleStyle={style.btn_text} type="outline" onPress={handleChangeCountMinus} disabled={formik.values[id]?.count === 1} />
          <Button id='plus' title='+' buttonStyle={style.btn} titleStyle={style.btn_text} type="outline" onPress={handleChangeCountPlus} disabled={formik.values[id]?.count >= max} />
        </View>
      </View>
    </View>
  )
}