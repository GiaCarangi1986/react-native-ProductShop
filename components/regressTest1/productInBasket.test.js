import 'jsdom-global/register';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { ProductItem } from './ProductItem'
import { mount, configure } from 'enzyme';

configure({ adapter: new Adapter() });
test('регрессионный тест обновления кол-ва продуктов в корзине', () => {
  const mockSetFieldValue = jest.fn()

  const title = 'Вологодское молоко 3.2%'
  const cost = 55
  const unit = 'р'
  const key = '4HbBsNdV8sB6Dmk78coH'
  const max = 10
  const formik = {
    values: {
      '4HbBsNdV8sB6Dmk78coH': {
        id: false,
        count: 2,
      }
    },
    setFieldValue: mockSetFieldValue,
  }
  const topRadius = false
  const bottomRadius = false
  const mockUpdateProduct = jest.fn()
  const id = '4HbBsNdV8sB6Dmk78coH'

  const component = mount(
    <ProductItem
      title={title}
      cost={cost}
      unit={unit}
      key={key}
      id={id}
      max={max}
      formik={formik}
      topRadius={topRadius}
      bottomRadius={bottomRadius}
      updateProduct={mockUpdateProduct}
    />
  )

  expect(component).toMatchSnapshot();

  component.find('#minus').simulate('click')
  component.find('#plus').simulate('click')

  expect(mockSetFieldValue).toBeCalledTimes(2)
  expect(mockUpdateProduct).toBeCalledTimes(2)
});