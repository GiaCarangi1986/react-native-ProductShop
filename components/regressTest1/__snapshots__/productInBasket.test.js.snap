// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`регрессионный тест обновления кол-ва продуктов в корзине 1`] = `ReactWrapper {}`;
